File: day01/day01.py
## ################################## ##
#### ### ### Advent of Code ### ### ####
# www.github.com/TeeJizzm/AdventOfCode #
## ################################## ##

## Day 01

############################
# Imports

import os

import tools.texttolists as tl

import tools.sumlist as sl

############################
# Variables



############################
# Functions

def day01(text):
    print("Day 01 - Calorie Counting")

    elves = tl.toList(text, "\n\n")
    elves = [elf.splitlines() for elf in elves]

    calories = []

    # Calculate the total calories carried by each elf
    for elf in elves:
        
        calorie = [int(item) for item in elf]
        calories.append(sum(calorie))

        #print(calorie)
    
    part1 = sl.sumTopList(calories)
    part2 = sl.sumTopList(calories, 3)

    return part1, part2

############################
# Run main

if __name__ == "__main__":
    day = "day01/inc" 
    
    # Change file
    #######
    #file = "ex.txt"
    file = "in.txt"
    #######

    # Get absolute filepath of file
    filepath = os.path.join(os.getcwd(), day, file)
    
    # Open file, clean up memory after
    with open(filepath, "r") as file:
        
        text = file.read() # Read data

    part1, part2 = day01(text)

    print("Part 1: ", part1)
    print("Part 2: ", part2)
    
########### EOF ############

File: day01/day01.go
/* ################################## ##
#### ### ### Advent of Code ### ### ####
# www.github.com/TeeJizzm/AdventOfCode #
## ################################## */

// Day 01

// -----------------------------------

package main

// Imports
import (
	// Input/Output
	"fmt"
	"io/ioutil"

	// String manipulation
	"strconv"
	"strings"

	// Day specific
	"sort"
)

/* -----------------------------------
// Functions */

func main() {
	fmt.Println("Day 01 - Calorie Counting")

	// ------------ File -------------

	//*/ - File switch
	file, err := ioutil.ReadFile("day01/inc/ex.txt")
	/*/ //
	file, err := ioutil.ReadFile("day01/inc/in.txt")
	// --- End --- */

	if err != nil {
		fmt.Println(err)
	}

	// Added fix for Windows newline
	text := strings.ReplaceAll(string(file), "\r\n", "\n")
	//fmt.Print(text)

	// ----------- Setup -------------

	elves := strings.Split(text, "\n\n")

	cals := make([]int, len(elves))

	for j, elf := range elves {
		cals[j] = 0

		items := strings.Fields(elf)

		for _, cal := range items {
			if cal != "" {

			}
			i, err := strconv.Atoi(cal)
			if err != nil {
				panic(err)
			}
			cals[j] += i
		}
	}

	//fmt.Println(cals)

	// ----------- Output ------------

	fmt.Println("Part 1:", sumTop(cals, 1))
	fmt.Println("Part 2:", sumTop(cals, 3))

} // End main

// -----------------------------------

func sumTop(array []int, num int) int {
	srt := array[:]
	sort.Ints(srt)
	total := 0

	for i := 1; i <= num; i++ {
		total += srt[len(srt)-i]
	}
	return total
}
